mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -185,7 +150,9 @@ computeCopyPerThread(Type elementType, int64_t copyPerThread, int64_t kPerBlock,
 static FailureOr<Value> wrapLDSBufferForStore(OpBuilder &b, Location loc,
                                               Value buffer, Type ldsReadType,
                                               int64_t kOuter, StringRef dName,
-                                              int64_t d, int64_t kPerThread) {
+                                              int64_t d, int64_t kPerThread,
+                                              int64_t dPerThread,
+                                              bool isKContiguousDim) {
```

Comment:
'isKContiguousDim' is going to be rather confusing when I call wrapLDSBufferForStore for the seccond gemm. I think 'rotateKwithD' is better because it gives the sense it is optional/perf improvement flag that called can use as long as blockwiseGemm knows that.

 With that I feel its better to name for blockwiseGemm 's new attrbitues to indicate K + D rotation is done on the input rather than saying it was done because K was contigous dim -- which is true but that should only be present in GridwiseGemmtoBlockwise where we make the decision to rotateKwithD when the input tiles for A & B is loaded from global memory.

Reply:
> Now I think about it again, I feel we should only do this if global kVecLen > 1.

Can we talk more about that? If I understand what is the issue you see, I am not sure kVecLen>1 is neither a good condition.

Reply:
We clarified this point with a chat :) Marking this as resolved